lib/eco_service_web/live/eco_service_live/community_list.ex

```diff
@@ -4,45 +4,73 @@ defmodule EcoServiceWeb.EcoServiceLive.CommunityList do
   alias EcoService.EcoServiceContext
 
   def mount(_params, _session, socket) do
-    params = %{limit: 10, offset: 0}
-    all_communities = EcoServiceContext.fetch_all_communities(params)
+    pag_params = %{limit: 10, offset: 0}
+    all_communities = EcoServiceContext.fetch_all_communities(pag_params)
 
     {:ok,
     socket
     |> assign(:communities, all_communities)
-    |> assign(:limit, params.limit)
-    |> assign(:offset, params.offset)
-    |> assign(:all_records, Enum.count(EcoServiceContext.fetch_all_communities()))
+    |> assign(:limit, pag_params.limit)
+    |> assign(:offset, pag_params.offset)
+    |> assign(:count_of_all_records, Enum.count(EcoServiceContext.fetch_all_communities()))
     }
   end
 
   def handle_event("prev", _params, socket) do
-    params= %{limit: 10,  offset: socket.assigns.offset-10}
-    all_communities = EcoServiceContext.fetch_all_communities(params)
+    pag_params= %{limit: 10,  offset: socket.assigns.offset-10}
```

Comment:
Why're you setting limit to 10 once again? That is already done in `mount/3` callback. Can you please let me know about a way where you need not re-set the values?

Please avoid using magic numbers in your code. Can you please let me know what is the magic number in the above line of code?

Reply:
I should have used socket.assigns.limit instead of reassigning it in the handle event.

The magic number is 10 on the above line.